decidim-proposals/app/controllers/concerns/decidim/proposals/admin/filterable.rb

```diff
@@ -47,10 +48,26 @@ def filters_with_values
               is_emendation_true: %w(true false),
               state_eq: proposal_states,
               scope_id_eq: scope_ids_hash(scopes.top_level),
-              category_id_eq: category_ids_hash(categories.first_class)
+              category_id_eq: category_ids_hash(categories.first_class),
+              valuator_role_ids_eq: valuator_role_ids
             }
           end
 
+          # Can't user `super` here, because it does not belong to a superclass
+          # but to a concern.
+          def dynamically_translated_filters
+            [:scope_id_eq, :category_id_eq, :valuator_role_ids_eq]
+          end
+
+          def valuator_role_ids
+            current_participatory_space.user_roles(:valuator).pluck(:id)
+          end
+
+          def translated_valuator_role_ids_eq(valuator_role_id)
+            user_role = current_participatory_space.user_roles(:valuator).find_by(id: valuator_role_id)
+            user_role.try(:user).try(:name)
```

Comment:
Just to be sure, would `user_role&.user&.name` do the same than this?

There is another possible value for `user_role` than `nil` or a `ParticipatorySpaceUserRole` instance?

Reply:
I just wanted to make sure the user actually existed, in case the data was wrong in the DB (a user role existing, but the related user being deleted somehow)

Reply:
Ok, I don't know if there is any preference between `try` and `&.`. I always thought that `&.` was for "I don't know if I will find a `nil` here", while `try` was more for "I don't know it this object will have this method". 

Yes, I know, its a very small difference... 😅

Reply:
I'll move to `&.` syntax!